From 2e152e003b8538ca73b61931837c9ce932b386ab Mon Sep 17 00:00:00 2001
From: Ville Vesilehto <ville@vesilehto.fi>
Date: Thu, 30 Jul 2026 11:38:22 +0300
Subject: [PATCH] fix: build on macOS where size_t is unsigned long

threefry() takes its counter as uint64_t&, but fill_rand()
declared offset_rounds as size_t. On macOS size_t is unsigned
long, which is a distinct type from uint64_t (unsigned long
long) even though both are 64 bits wide, so the reference
binding fails to compile with clang. Declare offset_rounds as
uint64_t instead.

Signed-off-by: Ville Vesilehto <ville@vesilehto.fi>
---
 util/Random.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/util/Random.cpp b/util/Random.cpp
index 1821ee93..53336d3a 100644
--- a/util/Random.cpp
+++ b/util/Random.cpp
@@ -495,9 +495,9 @@ static void fill_rand( uint8_t * out, const size_t elem_sz, const uint64_t elem_
     const size_t bytes_per_fill = Rand::BUFLEN * sizeof(uint64_t);
     uint8_t      tmp[Rand::BUFLEN * sizeof(uint64_t)];
 
-    size_t nbytes        = (elem_hi - elem_lo) * elem_sz;
-    size_t offset_rounds = (elem_lo * elem_sz) / (sizeof(uint64_t) * Rand::RANDS_PER_ROUND);
-    size_t offset_bytes  = (elem_lo * elem_sz) % (sizeof(uint64_t) * Rand::RANDS_PER_ROUND);
+    size_t   nbytes        = (elem_hi - elem_lo) * elem_sz;
+    uint64_t offset_rounds = (elem_lo * elem_sz) / (sizeof(uint64_t) * Rand::RANDS_PER_ROUND);
+    size_t   offset_bytes  = (elem_lo * elem_sz) % (sizeof(uint64_t) * Rand::RANDS_PER_ROUND);
     size_t offset_size   = std::min(sizeof(tmp) - offset_bytes, nbytes) % bytes_per_fill;
 
     if (offset_size > 0) {
-- 
2.50.1 (Apple Git-155)

